libc/x/paginate.c

```diff
@@ -17,42 +17,93 @@
 │ PERFORMANCE OF THIS SOFTWARE.                                                │
```

Comment:
Why move this from `libc/proc` to `libc/x`?

Reply:
> paginate.c was moved from libc/proc to libc/x mainly to get utf16to8. However, it seems to fit as pagination is not standard, but is implemented in a variety of codebases.

I think it needs to move to get access to `utf16to8` without `_weaken`. The other justification refers to the description of libc/x in it's `BUILD.MK`.

Reply:
Per discord, suggesting keeping this in `libc/proc` and using `_weaken`.

Reply:
I moved it back to `libc/proc` and switched it to use `_weaken` in 38d0728, however the `__static_yoink`  doesn't work from there, in order to get `utf16to8` linked in I had to  `__static_yoink("__utf16to8");` directly from my `__paginate` using program.

Any suggestions of how I could proceed to avoid requiring explicit linking of `__utf16to8` ? I could wrap `tprecode16to8` and use `strlen16` and `malloc` instead, but at that point we're reinventing `utf16to8` inside of `libc/proc`.

Reply:
ty for the poke - I had missed this.

I do not have a smart answer at the moment. It seems like the three dumb options are:

1. push the dependency off onto the user as you indicate;
2. put the yoink in the file in `libc/proc` thus creating a dependency on that function out of `libc/x` from `libc/proc`; or 
3. reimplement it in `libc/proc`.

My knowledge of what `__static_yoink` is doing has paged out so I'm not sure if there are any problems created by option 2. If you want this merged quickly and are okay with making fixes and refinements later, I would say go with option 1.

Reply:
I think I messed up by focusing on the move to `libc/x`. 

Looking at the root Makefile, we see that `LIBC_PROC` is declared before `LIBC_X`, and IIRC the parts with fewer dependencies (`LIBC_PROC` in this case) are declared before `LIBC_X`.

So having something in `libc/proc` depend on `libc/x` would introduce some circular issue? Perhaps the easiest way for now is to keep it in `LIBC_X`.